application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,213 @@
+     */
+    public TransferQuestionCommand() {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+
+    }
+
+    /**
+     * Verifies if user has role higher or equal to Community Ambassador in terms of hierarchy.
+     */
+    public boolean hasRequiredRoles(MessageContextInteractionEvent event) {
+
+        List<Role> userRoles = event.getMember().getRoles();
+        List<Role> communityAmbassador =
+                event.getGuild().getRolesByName("COMMUNITY AMBASSADOR", true);
+
+        for (Role role : userRoles) {
+            if (role.getPosition() >= communityAmbassador.get(0).getPosition()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+
+        if (!hasRequiredRoles(event)) {
+            event.reply("You're not authorised to use this command").setEphemeral(true).queue();
+        }
+
+        String originalMessage = event.getTarget().getContentRaw();
+
+
+        originalMessageId = event.getTarget().getId();
+        originalChannelId = event.getChannel().getId();
+
+        String author = event.getTarget().getAuthor().getId();
+
+
+        TextInput transferQuestionTitle =
+                TextInput.create(TRANSFER_QUESTION_TITLE_ID, "Title", TextInputStyle.SHORT)
+                    .setMaxLength(70)
+                    .setMinLength(8)
+                    .setValue(createTitle(originalMessage))
+                    .build();
+
+        TextInput transferQuestionInput = TextInput
+            .create(TRANSFER_QUESTION_INPUT_ID, "Transfer question menu", TextInputStyle.PARAGRAPH)
+            .setValue(originalMessage)
+            .setRequiredRange(3, 1000)
+            .build();
+
+        TextInput transferQuestionTag = TextInput
+            .create(TRANSFER_QUESTION_TAG, "Transfer question tags", TextInputStyle.SHORT)
+            .setValue("Java")
+            .build();
+
+
+
+        String transferQuestionModalComponentID = generateComponentId(author);
+        Modal reportModal = Modal.create(transferQuestionModalComponentID, "transfer question menu")
+            .addActionRow(transferQuestionTitle)
+            .addActionRow(transferQuestionInput)
+            .addActionRow(transferQuestionTag)
+            .build();
+
+        event.replyModal(reportModal).queue();
+    }
+
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        User author = event.getJDA().getUserById(args.get(0));
+        createForumPost(event, author);
+
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
```

Comment:
use util  method

Reply:
which one? This is something i used from earlier help system to generate title

Reply:
[MessageUtils#abbreviate](https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/features/utils/MessageUtils.java#L148)

Reply:
that method is a different one with a different behavior and purpose

Reply:
abbreviate cuts in the middle of a word and then adds an elipsis. like `hello wo...`.
the one bee is using here is the old code that we used for the old help system, for guessing thread titles. it cuts at word boundaries instead.
subtle difference, but better UX.

it would make sense though to offer it as helper somewhere